Feature/event hosts

Kneeba921 opened the pull request:
- adds event_host migration and model
- an event can now accept nested attributes for its host
- updates to the event create form to accept and pass its host type and id

rsmithlal commented on app/models/concerns/better_together/hosts_events.rb:
```diff
@@ -0,0 +1,20 @@
+module BetterTogether
+  # Concern that when included gives the model access to events through event_host records
+  # This module must be included in a model to permit assigning instances as an event host
+  module HostsEvents
```
@Kneeba921 include this module in https://github.com/NL-Venues/nl-venues/blob/feature/event-hosts/app/models/venue.rb to allow-list the class as a permitted event host type

rsmithlal commented:
I did a little refactor here and added the ability to create events for a community, too. Please review my commits to see what I changed, and add comments to changed lines if you'd like me to explain what I did or why.

Kneeba921 commented:
> I did a little refactor here and added the ability to create events for a community, too. Please review my commits to see what I changed, and add comments to changed lines if you'd like me to explain what I did or why.

I think I'm pretty good with it. Still maybe a little iffy with the scoping in the policies, but I get the gist of it.
